packages/postgres-database/src/simcore_postgres_database/utils_folders.py

```diff
@@ -415,6 +425,53 @@ def _get_permissions_where_clause() -> ColumnElement | bool:
 
 
 async def _check_folder_and_access(
+    connection: SAConnection,
+    product_name: _ProductName,
+    folder_id: _FolderID,
+    gids: set[_GroupID],
+    *,
+    permissions: _FolderPermissions,
+    enforece_all_permissions: bool,
+) -> _ResolvedAccessRights:
+    """
+    Raises:
+        FolderNotFoundError
+        FolderNotSharedWithGidError
+        InsufficientPermissionsError
+        UnexpectedFolderAccessError
+    """
+    resolved_access_rights: _ResolvedAccessRights | None = None
+    last_exception: Exception | None = None
+    for gid in gids:
+        try:
+            resolved_access_rights = await _check_folder_access(
+                connection,
+                product_name,
+                folder_id,
+                gid,
+                permissions=permissions,
+                enforece_all_permissions=enforece_all_permissions,
```

Comment:
```suggestion
                enforce_all_permissions=enforce_all_permissions,
```

Reply:
Also, what is the use case where you check the access without "enforcing all permissions"?